BIPs/2023-W22/BIP-315.json

```diff
@@ -0,0 +1,100 @@
+{
+  "version": "1.0",
+  "chainId": "1",
+  "createdAt": 1685636319821,
+  "meta": {
+    "name": "Transactions Batch",
+    "description": "",
+    "txBuilderVersion": "1.14.1",
+    "createdFromSafeAddress": "0x10A19e7eE7d7F8a52822f6817de8ea18204F2e4f",
+    "createdFromOwnerAddress": "",
+    "checksum": "0xe0a2a3143ed114acf0885ab3f17ca9072fb545d6c5540b4331606075663a6309"
+  },
+  "transactions": [
+    {
+      "to": "0xA331D84eC860Bf466b4CdCcFb4aC09a1B43F3aE6",
+      "value": "0",
+      "data": null,
+      "contractMethod": {
+        "inputs": [
+          { "internalType": "bytes32[]", "name": "roles", "type": "bytes32[]" },
+          { "internalType": "address", "name": "account", "type": "address" }
+        ],
+        "name": "grantRoles",
+        "payable": false
+      },
+      "contractInputsValues": {
```

Comment:
If approved, the DAO Multisig 0x10A19e7eE7d7F8a52822f6817de8ea18204F2e4f will initiate a transaction to interact with the Authorizer 0xA331D84eC860Bf466b4CdCcFb4aC09a1B43F3aE6 calling grantRoles with the following arguments:

roles: [0x2960d085c4c968d7cad55c0da3f97014525b948fdce990ecaef4e832b5f0b151,0x3f44776af02a9227991da37715b44e45db40735e216b3ab33144859bb6737166]

These correspond to the roles for calling addGaugeType and setGaugeFactory on the Gauge Adder v4, verifiable [here](https://github.com/balancer/balancer-deployments/blob/master/action-ids/mainnet/action-ids.json#L1342).

address: 0x10A19e7eE7d7F8a52822f6817de8ea18204F2e4f

This is the address for the DAO Multisig to allow it to call the above functions.